src/constraint_utils.R

```diff
@@ -0,0 +1,47 @@
+# LT 18/05/2022
+
+# utility function for 3D missense constraint metric
+
+## required packages
+# for .PDB structures
+require(bio3d)
+
+# set seed for reproducibility
+set.seed(9011971)
+
+## pairwise distances between residues based on their carbon alpha
+get_dist_mat <- function(pdb_file) {
+  pdb <- read.pdb(pdb_file)
+
+  # calculate distance between residues based on their carbon alpha
+  dist_mat <- dm(pdb, inds = "calpha")
+
+  # dist_mat is upper triangular with empty diagonal: add diagonal, fill lower triangle, convert to matrix
+  dist_mat <- as.matrix(dist_mat)
+  dist_mat[is.na(dist_mat)] <- 0
+  dist_mat <- dist_mat + t(dist_mat)
+
+  dist_mat
+}
+
+## export a dataset containing intolerance scores in a format compatible with
+## iCn3D
+
+export_iCn3D <- function(oe_data, file) {
+  if ("numeric" %in% class(oe_data)) {
+    oe <- oe_data
+  } else {
+    oe <- oe_data[, "oe"]
+  }
+
+  # set score on 0-100 range (needed by iCn3D)

```

Comment:
just a comment, nothing to action:
re: allowing 0-100 and no NA values, I think this is also a quirk with the other protein viewers as well that we might need to factor in later, though potentially more annoying if we can't use negative numbers.

So for residues we don't want coloured we can normally set the score as -1, then in a viewer select residues with -1 and colour something neutral. If we can't use negatives we might need to handle it differently